From c561e62d908c992b18fb5c050b4489ee979f4351 Mon Sep 17 00:00:00 2001
From: Ankit Agrawal <ankit.agarwal@intel.com>
Date: Mon, 19 Jun 2023 09:55:51 +0530
Subject: [PATCH] Avoiding Error logs During framework boot.

Currently We are seeing continously below error logs due to
drop box manager service, since it is not published yet by
system server.

Avoid error logs by adding null check before calling getSystemService.

error logs:
E SystemServiceRegistry: No service published for: dropbox
E SystemServiceRegistry: android.os.ServiceManager$ServiceNotFoundException: No service published for: dropbox
E SystemServiceRegistry:        at android.os.ServiceManager.getServiceOrThrow(ServiceManager.java:166)
E SystemServiceRegistry:        at android.app.SystemServiceRegistry$25.createService(SystemServiceRegistry.java:491)
E SystemServiceRegistry:        at android.app.SystemServiceRegistry$25.createService(SystemServiceRegistry.java:488)
E SystemServiceRegistry:        at android.app.SystemServiceRegistry$CachedServiceFetcher.getService(SystemServiceRegistry.java:1971)
E SystemServiceRegistry:        at android.app.SystemServiceRegistry.getSystemService(SystemServiceRegistry.java:1645)
E SystemServiceRegistry:        at android.app.ContextImpl.getSystemService(ContextImpl.java:2212)
E SystemServiceRegistry:        at android.content.Context.getSystemService(Context.java:4420)
E SystemServiceRegistry:        at com.android.server.am.ActivityManagerService.addErrorToDropBox(ActivityManagerService.java:9146)
E SystemServiceRegistry:        at com.android.server.am.ActivityManagerService.handleApplicationWtfInner(ActivityManagerService.java:8963)
E SystemServiceRegistry:        at com.android.server.am.ActivityManagerService$15.run(ActivityManagerService.java:8929)
E SystemServiceRegistry:        at android.os.Handler.handleCallback(Handler.java:958)
E SystemServiceRegistry:        at android.os.Handler.dispatchMessage(Handler.java:99)
E SystemServiceRegistry:        at android.os.Looper.loopOnce(Looper.java:205)
E SystemServiceRegistry:        at android.os.Looper.loop(Looper.java:294)
E SystemServiceRegistry:        at android.os.HandlerThread.run(HandlerThread.java:67)
E SystemServiceRegistry:        at com.android.server.ServiceThread.run(ServiceThread.java:46)
E SystemServiceRegistry: Manager wrapper not available: dropbox

Tracked-On: OAM-110091
Change-Id: Id954ecf8dd94831576e0a329f27231631124bada
Signed-off-by: Ankit Agrawal <ankit.agarwal@intel.com>
---
 .../com/android/server/am/ActivityManagerService.java     | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/services/core/java/com/android/server/am/ActivityManagerService.java b/services/core/java/com/android/server/am/ActivityManagerService.java
index 48898a6393db..6c72a8c6689f 100644
--- a/services/core/java/com/android/server/am/ActivityManagerService.java
+++ b/services/core/java/com/android/server/am/ActivityManagerService.java
@@ -9141,12 +9141,8 @@ public class ActivityManagerService extends IActivityManager.Stub
         // otherwise the watchdog may be prevented from resetting the system.
 
         // Bail early if not published yet
-        final DropBoxManager dbox;
-        try {
-            dbox = mContext.getSystemService(DropBoxManager.class);
-        } catch (Exception e) {
-            return;
-        }
+        if (ServiceManager.getService(Context.DROPBOX_SERVICE) == null) return;
+        final DropBoxManager dbox = mContext.getSystemService(DropBoxManager.class);
 
         // Exit early if the dropbox isn't configured to accept this report type.
         final String dropboxTag = processClass(process) + "_" + eventType;
-- 
2.17.1

